src/scheduler.c

```diff
@@ -1,340 +1,6 @@
-    shutdown_tag = self->stop_tag;
-  }
-
-  self->do_shutdown(self, shutdown_tag);
-}
-
-lf_ret_t Scheduler_schedule_at_locked(Scheduler *self, Event *event) {
-  // Check if we are trying to schedule past stop tag
-  if (lf_tag_compare(event->tag, self->stop_tag) > 0) {
-    LF_WARN(SCHED, "Trying to schedule trigger %p at tag %" PRId64 ":%" PRIu32 " past stop tag %" PRId64 ":%" PRIu32,
-            event->trigger, event->tag.time, event->tag.microstep, self->stop_tag.time, self->stop_tag.microstep);
-    return LF_AFTER_STOP_TAG;
-  }
-
-  // Check if we are tring to schedule into the past
-  if (lf_tag_compare(event->tag, self->current_tag) <= 0) {
-    LF_WARN(SCHED,
-            "Trying to schedule trigger %p at tag %" PRId64 ":%" PRIu32 " which is before current tag %" PRId64
-            ":%" PRIu32,
-            event->trigger, event->tag.time, event->tag.microstep, self->current_tag.time, self->current_tag.microstep);
-    return LF_PAST_TAG;
-  }
-
-  lf_ret_t ret = self->event_queue.insert(&self->event_queue, event);
-  if (ret != LF_OK) {
-    LF_ERR(SCHED, "Failed to insert event into event queue");
-  }
-
-  return ret;
-}
-
-lf_ret_t Scheduler_schedule_at(Scheduler *self, Event *event) {
-  Environment *env = self->env;
-
-  env->enter_critical_section(env);
-
-  int res = self->schedule_at_locked(self, event);
-
-  env->leave_critical_section(env);
-
-  return res;
-}
-
-void Scheduler_set_duration(Scheduler *self, interval_t duration) { self->duration = duration; }
-
-void Scheduler_request_shutdown(Scheduler *self) {
-  Environment *env = self->env;
-  env->enter_critical_section(env);
-  self->stop_tag = lf_delay_tag(self->current_tag, 0);
-  LF_INFO(SCHED, "Shutdown requested, will stop at tag %" PRId64 ":%" PRIu32, self->stop_tag.time,
-          self->stop_tag.microstep);
-  env->platform->new_async_event(env->platform);
-  env->leave_critical_section(env);
-}
-
-void Scheduler_ctor(Scheduler *self, Environment *env) {
-  self->env = env;
-  self->run = Scheduler_run;
-  self->prepare_timestep = Scheduler_prepare_timestep;
-  self->clean_up_timestep = Scheduler_clean_up_timestep;
-  self->run_timestep = Scheduler_run_timestep;
-  self->do_shutdown = Scheduler_do_shutdown;
-  self->schedule_at = Scheduler_schedule_at;
-  self->schedule_at_locked = Scheduler_schedule_at_locked;
-  self->register_for_cleanup = Scheduler_register_for_cleanup;
-  self->request_shutdown = Scheduler_request_shutdown;
-  self->acquire_and_schedule_start_tag = Scheduler_acquire_and_schedule_start_tag;
-  self->keep_alive = false;
-  self->stop_tag = FOREVER_TAG;
-  self->current_tag = NEVER_TAG;
-  self->start_time = NEVER;
-  self->duration = FOREVER;
-  self->cleanup_ll_head = NULL;
-  self->cleanup_ll_tail = NULL;
-  self->leader = false;
-  EventQueue_ctor(&self->event_queue);
-  ReactionQueue_ctor(&self->reaction_queue);
-}
+void Scheduler_ctor(Scheduler *self) { (void)self; }
```

Comment:
I think here we could put the #ifdefs like for platform, `Scheduler_ctor` can probably just be removed?